calculation_scripts/OptimizationWithWrapper.py

```diff
@@ -0,0 +1,92 @@
+import dda_model
+from dda_model import optimizers
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+def construct_model():
+    sym_axis = [10.5, 10.5]
+    geometry_shape = [22, 22, 10]
+    pixel_size = 15.0
+    light_direction = [0, 0, 1]
+    light_polarization = [1, 0, 0]
+    wavelength = 542
+    initialization = np.loadtxt("initializations/hourglass.txt")
+    dielectric_constants = [1.01 + 0j, 5.96282 + 3.80423e-7j]
+
+    model = dda_model.DDAModelWrapper(
+        geometry_shape,
+        pixel_size,
+        initialization,
+        dielectric_constants=dielectric_constants,
+        light_direction=light_direction,
+        light_polarization=light_polarization,
+        light_wavelength_nm=wavelength,
+        symmetry_axes=sym_axis,
+        verbose=True,
+    )
+    return model
+
+def saveCurrentStructure(model_, path, iteration):
+    all_parameters = model_.allParameters()
+
+    with open(path + '\\CoreStructure\\CoreStructure' + str(iteration) + '.txt', 'w') as f:
+        for para in all_parameters:
+            f.write(f"{para}\n")
+
+def plotObjectiveFunction(objective_values, path):
+    plt.figure(1)
```

Comment:
Since this is within a function, might be better to just work on the currently-active figure rather than make a new one. Also, `figure(1)` will always update the same figure (the one with ID = 1). So if you call this multiple times, it'll always plot on the same underlying figure.

Reply:
how do you activate a currently-active figure? I thought this was done using plt.figure(). But yes, setting to plt.figure(1) was causing overlap of my plots onto each other lol. For now, I just deleted that line, which seems to be working alright. But I might want to revisit this with you

Reply:
plt.figure() will make a new figure and set the currently-active figure to the new figure.

plt.figure(N) will look for figure N. If it doesn't exist, it'll make a new figure. Then it sets the currently active figure to figure N.

Reply:
Gotcha. Thanks!